### PR TITLE
[WS-pscan-scripts_js] Base64 Disclosure

### PR DESCRIPTION
**Pull request by eakirtas**

### CODE REVIEW

File: addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/ExtensionWebSocket.java [hunk 1]
```diff
@@ -525,6 +537,62 @@ public String getDescription() {
...
+                    .map(maybeScriptWrapper -> maybeScriptWrapper.get())
+                    .filter(
+                            scriptWrapper ->
+                                    this.extensionScript.getScript(scriptWrapper.getName()) == null)
+                    .forEach(scriptWrapper -> this.extensionScript.addScript(scriptWrapper, false));
+
+        } catch (NoSuchFileException e) {
+            return;
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (kingthorin) [inline addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/ExtensionWebSocket.java:593]: This should use a proper logger.

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/base64_disclosure.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                print("Decoded Evidence: " + decodedEvidence);
+            }
+
+            helper.newAlert()
+                .setRiskConfidence(RISK_INFO, CONFIDENCE_MEDIUM)
+                .setName("Base64 Disclosure in WebSocket message (script)")
+                .setDescription("A Base64-encoded string has been founded in the websocket incoming message. Base64-encoded data may contain sensitive" +
+                                "information such as usernames, passwords or cookies which should be further inspected. Decode evidence: "
+                                + decodedEvidence + ".")
+                .setParam(counter++)
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/base64_disclosure.js]: I think this is probably used on some handling in the [Find HTML Comments](https://github.com/zaproxy/community-scripts/blob/27efdf1cb39dee4f9dd71d7055e4b5a6ee231c42/passive/Find%20HTML%20Comments.js#L23-L24) script. However: 1) It's an older script. 2) It uses parameters in order to facilitate different grouping of results.

In the majority of cases passive scanners (scripts or other) should not really have/need a "parameter".

File: addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/ExtensionWebSocket.java (hunk 1 above)
Comment (eakirtas) [reply addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/ExtensionWebSocket.java:593]: Fixed in a3e8c743a

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/base64_disclosure.js (hunk 2 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/base64_disclosure.js]: Fixed in f5d80fa21

**Pull request closed without merging**